fury/actor.py

```diff
@@ -754,6 +754,103 @@ def callback(_caller, _event, calldata=None):
     return actor
 
 
+def dashed_line(start_pos, end_pos, colors=np.array([[1, 0, 0]]),
+                num_lines=10, line_fraction=0.5):
+    """Create an actor for one or more dashed lines.
+
+    Parameters
+    ------------
+    start_pos :  array (N, 3)
+    end_pos :  array (N, 3)
+    num_lines :  int, optional
+    colors :  array (N, 3)
+    line_fraction :  float
+         it should be between 0 and 1
+
+    Returns
+    ----------
+    v : vtkActor
+        Dashed Line.
+
+    Examples
+    ----------
+    >>> from fury import window, actor
+    >>> import numpy as np
+    >>> scene = window.Scene()
+    >>> start_pos = np.random.rand(3, 3)
+    >>> end_pos = np.random.rand(3, 3)
+    >>> num_lines = 20
+    >>> colors = np.random.rand(3, 3)
+    >>> line_fraction = 0.5
+    >>> c = actor.dashed_line(start_pos, end_pos, colors,
+    >>> num_lines, line_fraction)
+    >>> scene.add(c)
+    >>> #window.show(scene)
+    """
+    N = start_pos.shape[0]
+    arr = np.empty((num_lines * N, 2, 3))
+
+    for start_pos, end_pos, x in zip(start_pos, end_pos, range(0, N)):
+        dp = (end_pos - start_pos) / num_lines
+        p = start_pos.copy()
+        count = 0
+        while count < num_lines:
+            arr[x * num_lines + count] = \
+                np.array([p,  p + (dp * line_fraction)])
+            count = count + 1
+            p = p + dp
+    c = line(arr, np.repeat(colors, num_lines, axis=0))
+    return c
+
+
+def dotted_line(start_pos, end_pos, colors=np.array([[1, 0, 0]]),
+                num_points=10, radius=0.01):
+    """Create an actor for one or more dotted lines.
+
+    Parameters
+    ------------
+    start_pos : array (N, 3)
+    end_pos :  array (N, 3)
+    num_points :  int, optional
+    colors :  array (N, 3)
+    radius :  float
+
+    Returns
+    ----------
+    v : vtkActor
+        Dotted Line.
+
+    Examples
+    ----------
+    >>> from fury import window, actor
+    >>> import numpy as np
+    >>> scene = window.Scene()
+    >>> start_pos = np.random.rand(3, 3)
+    >>> end_pos = np.random.rand(3, 3)
+    >>> num_points = 20
+    >>> colors = np.random.rand(3, 3)
+    >>> radius = 0.01
+    >>> c = actor.dotted_line(start_pos, end_pos, colors,
+    >>> num_points,radius)
+    >>> scene.add(c)
+    >>> #window.show(scene)
+    """
+    N = start_pos.shape[0]
+    arr = np.empty((num_points * N, 3))
+
+    for start_pos, end_pos, x in zip(start_pos, end_pos, range(0, N)):
+        dp = (end_pos - start_pos) / num_points
+        p = start_pos.copy()
+        count = 0
+        while count < num_points:
+            arr[x * num_points + count] = np.array([p])
+            count = count + 1
+            p = p + dp
+    c = point(points=arr, colors=np.repeat(colors, num_points, axis=0),
```

Comment:
Thank you @haran2001 for creating dotted line and dashed line. They are very useful. Here are my comments:
- It's better to have two options for creating actor in dotted_line:  "point" and "dots" actors. The point actor is slow because it is creating spheres with full geometry. People should be able to use ``actor.dots`` instead that does not generated 3D geometry. Also there is a fast sphere rendering option put that can be added at another PR.
- If num_points=2, it will create two points, the first point is located at start_pos, the second point is located in the middle of the line (not at end_pos). Here I showed two snapshots. To make sure if the second point is not located at end_pos, I created sphere actors for start_point and end_point with bigger radius. You will see that the second point is exactly in the middle of start_pos and end_pos. I think this should be corrected.
![image](https://user-images.githubusercontent.com/32108826/114611752-5cf82000-9c6f-11eb-928f-ff7cd73543e3.png)

Reply:
Resetting the end position is an simple fix. However adding the dots option is difficult as the dot actor is implemented through a **single unstructured grid**  in VTK . This makes it difficult to plot multiple dotted lines with different colors and all the line end up with the same color of the grid actor.

Reply:
Do you have any suggestions on how to approach this issue?

Reply:
The dots_actor parameter has been added, which uses fury dots to render the dotted line.